===================================================================
RCS file: /cvsroot/xchat/xchat2/src/fe-gtk/fkeys.c,v
retrieving revision 1.37
retrieving revision 1.38
diff -u -r1.37 -r1.38
--- xchat/xchat2/src/fe-gtk/fkeys.c	2004/07/30 16:06:16	1.37
+++ xchat/xchat2/src/fe-gtk/fkeys.c	2004/07/31 05:19:52	1.38
@@ -1368,9 +1368,14 @@
 	    prefix_len, skip_len = 0, is_nick, is_cmd = 0;
 	char buf[COMP_BUF], ent[CHANLEN], *postfix = NULL, *result, *ch;
 	GList *list = NULL, *tmp_list = NULL;
-	const char *text = gtk_entry_get_text (GTK_ENTRY (t));
+	const char *text;
 	GCompletion *gcomp = NULL;
 
+	/* force the IM Context to reset */
+	gtk_editable_set_editable (GTK_EDITABLE (t), FALSE);
+	gtk_editable_set_editable (GTK_EDITABLE (t), TRUE);
+
+	text = GTK_ENTRY (t)->text;
 	if (text[0] == 0)
 		return 1;
 
